### PR TITLE
Simple installer for novices

### PR DESCRIPTION
**Pull request by Shigbeard**
This addresses the issues that were outlined in #23. The SQL is imported
automatically, the database file is copied across from the sample and
modified using str_replace. Same with the config.

Index page redirects to install file in the event that the database file
cannot be found. Also added some extra security against naughty bots
that ignore headers, or dodgy pre HTTP/1.1 user agents (all modern
browsers should meet HTTP/1.1 standards)

EDIT: Forgot i did not author #23. Pronouns corrected.

### CODE REVIEW

File: install.php [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+}
+
+function requestInfo()
+{
+	// This is the way novices will be installing the script.
+	// That is why I only ask them for critical details, the rest can remain default.
+	echo <<<HTML
+		<div class="container">
+			<h1>Setup</h1>
+			<p>In order for this package to function, you must supply access details to a MySQL database, and a secret encryption key (this can be random).</p>
```
Comment (Arrow768) [inline install.php]: You should also note, that this database needs to contain the tables for the store plugin

File: install.php [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+HTML;
+}
+
+function processInfo() 
+{
+	// Import the SQL stuffs first. If these go wrong, no harm done.
+	// This is going to hurt your eyes.
+	mysql_connect($_POST['host'],$_POST['user'],$_POST['pass']) or die('Error connecting to MySQL. No harm done, just start the installation again.');
+	mysql_select_db($_POST['name']) or die('Error selecting MySQL database: ' . mysql_error());
+
```
Comment (Arrow768) [inline install.php]: You should perform a check if the store tables are already imported and throw a Info message if they are not.

(Just check if the store_users table exists)

File: install.php [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+	fclose($confighandler);
+	
+	$configfile = str_replace("[REPLACE-ENCRYPTION_KEY]", $_POST['key'], $configfile);
+	$realconfigfile = fopen('./application/config/config.php', 'w') or exit('Error creating Config file. Does the system have access to the directory?');
+	fwrite($realconfigfile, $configfile);	
+	echo "Config saved!<br>";
+	echo <<<HTML
+		<meta http-equiv="refresh" content="3;url=./" />
+		<b>Redirecting in 3 seconds...</b>
+HTML;
```
Comment (Arrow768) [inline install.php]: You should offer a link that users can click on if the redirect is not working

**Comment by Arrow768** [comment]
I added a few notes on how the installer could be improved to prevent a few common problems from happening. (Wrong db, Store Tables dont exist, Redirect not working)

Its already already a good job, but I would appreciate it if you could improve these points

**Comment by HDG-MrDeeJay** [comment]
Righto. Fair call on all those points. I'll get to work immediately.

**Pull request closed without merging**